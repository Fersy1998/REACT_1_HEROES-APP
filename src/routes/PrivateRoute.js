import React, { useContext } from 'react'
import { Navigate, useLocation } from 'react-router-dom';
import { AuthContext } from '../auth/authContext'



//Children son los componentes que envuelve PrivateRoute en AppRoutes
export const PrivateRoute = ({children}) => {

    //const location=useLocation();
    //desestructuramos y obtenemos el pathname y el search
    const {pathname, search}=useLocation();
    
    localStorage.setItem('last_path',pathname+search);
    const {user}=useContext(AuthContext);
    
    
  return user.logged ? children : <Navigate to='/login' />
            
}
