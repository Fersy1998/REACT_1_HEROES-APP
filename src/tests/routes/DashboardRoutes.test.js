import { DashboardRoutes } from "../../routes/DashboardRoutes";
import {mount} from 'enzyme'
import {AuthContext} from '../../auth/authContext'
import { MemoryRouter } from "react-router-dom";

describe('Pruebas en DashboardRoutes.js', ()=>{

   const contextValue={user:{
    logged:true,
    name:'Sara'
   }}
    test('should show correctly /search', () => { 
        const wrapper=mount(
        <AuthContext.Provider value={contextValue}>
        
            <MemoryRouter initialEntries={['/search']}>
                <DashboardRoutes />
            </MemoryRouter>
          
        </AuthContext.Provider>  );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('span').text().trim()).toBe('Sara');
    
    
    })
    test('should show correctly /marvel', () => { 
        const wrapper=mount(
        <AuthContext.Provider value={contextValue}>
  
            <MemoryRouter initialEntries={['/marvel']}>
                <DashboardRoutes />
            </MemoryRouter>
          
        </AuthContext.Provider>  );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text()).toContain('Marvel Comics');

    })
    test('should show correctly /dc', () => { 
        const wrapper=mount(
        <AuthContext.Provider value={contextValue}>
  
            <MemoryRouter initialEntries={['/dc']}>
                <DashboardRoutes />
            </MemoryRouter>
          
        </AuthContext.Provider>  );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text()).toContain('DC Comics');

    })
})