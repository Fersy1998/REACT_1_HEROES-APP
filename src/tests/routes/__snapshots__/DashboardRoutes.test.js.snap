// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DashboardRoutes.js should show correctly /dc 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-linkactive"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end container"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link"
        >
          Sara
        </span>
        <button
          className="nav-item nav-link btn btn-outline-warning"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <div
        className="row "
      >
        <h1>
          Hero List - 
          DC Comics
        </h1>
        <hr />
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-batman"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Batman"
              className="card-img-top"
              src="/assets/dc-batman.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Detective Comics #27
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-superman"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Superman"
              className="card-img-top"
              src="/assets/dc-superman.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Superman
              </h5>
              <p
                className="card-text"
              >
                Action Comics #1
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                Kal-El
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-superman"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-flash"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Flash"
              className="card-img-top"
              src="/assets/dc-flash.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Flash
              </h5>
              <p
                className="card-text"
              >
                Flash Comics #1
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                Jay Garrick
              </p>
              <p
                className="card-text text-muted"
              >
                Jay Garrick, Barry Allen, Wally West, Bart Allen
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-flash"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-green"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Green Lantern"
              className="card-img-top"
              src="/assets/dc-green.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Green Lantern
              </h5>
              <p
                className="card-text"
              >
                All-American Comics #16
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                Alan Scott
              </p>
              <p
                className="card-text text-muted"
              >
                Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-green"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-arrow"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Green Arrow"
              className="card-img-top"
              src="/assets/dc-arrow.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Green Arrow
              </h5>
              <p
                className="card-text"
              >
                More Fun Comics #73
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                Oliver Queen
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-arrow"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-wonder"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Wonder Woman"
              className="card-img-top"
              src="/assets/dc-wonder.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Wonder Woman
              </h5>
              <p
                className="card-text"
              >
                All Star Comics #8
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                Princess Diana
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-wonder"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-martian"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Martian Manhunter"
              className="card-img-top"
              src="/assets/dc-martian.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Martian Manhunter
              </h5>
              <p
                className="card-text"
              >
                Detective Comics #225
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                J'onn J'onzz
              </p>
              <p
                className="card-text text-muted"
              >
                Martian Manhunter
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-martian"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-robin"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Robin/Nightwing"
              className="card-img-top"
              src="/assets/dc-robin.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Robin/Nightwing
              </h5>
              <p
                className="card-text"
              >
                Detective Comics #38
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                Dick Grayson
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-robin"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-blue"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Blue Beetle"
              className="card-img-top"
              src="/assets/dc-blue.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Blue Beetle
              </h5>
              <p
                className="card-text"
              >
                Mystery Men Comics #1
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                Dan Garret
              </p>
              <p
                className="card-text text-muted"
              >
                Dan Garret, Ted Kord, Jaime Reyes
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-blue"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="dc-black"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Black Canary"
              className="card-img-top"
              src="/assets/dc-black.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Black Canary
              </h5>
              <p
                className="card-text"
              >
                Flash Comics #86
              </p>
              <p
                className="card-text"
              >
                DC Comics
              </p>
              <p
                className="card-text"
              >
                Dinah Drake
              </p>
              <p
                className="card-text text-muted"
              >
                Dinah Drake, Dinah Lance
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-black"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en DashboardRoutes.js should show correctly /marvel 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-linkactive"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end container"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link"
        >
          Sara
        </span>
        <button
          className="nav-item nav-link btn btn-outline-warning"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <div
        className="row "
      >
        <h1>
          Hero List - 
          Marvel Comics
        </h1>
        <hr />
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-spider"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Spider Man"
              className="card-img-top"
              src="/assets/marvel-spider.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Spider Man
              </h5>
              <p
                className="card-text"
              >
                Amazing Fantasy #15
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Peter Parker
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-captain"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Captain America"
              className="card-img-top"
              src="/assets/marvel-captain.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Captain America
              </h5>
              <p
                className="card-text"
              >
                Captain America Comics #1
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Steve Rogers
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-iron"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Iron Man"
              className="card-img-top"
              src="/assets/marvel-iron.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Iron Man
              </h5>
              <p
                className="card-text"
              >
                Tales of Suspense #39
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Tony Stark
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-thor"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Thor"
              className="card-img-top"
              src="/assets/marvel-thor.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Thor
              </h5>
              <p
                className="card-text"
              >
                Journey into Myster #83
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Thor Odinson
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-hulk"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Hulk"
              className="card-img-top"
              src="/assets/marvel-hulk.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hulk
              </h5>
              <p
                className="card-text"
              >
                The Incredible Hulk #1
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Bruce Banner
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-wolverine"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Wolverine"
              className="card-img-top"
              src="/assets/marvel-wolverine.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Wolverine
              </h5>
              <p
                className="card-text"
              >
                The Incredible Hulk #180
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                James Howlett
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-daredevil"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Daredevil"
              className="card-img-top"
              src="/assets/marvel-daredevil.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Daredevil
              </h5>
              <p
                className="card-text"
              >
                Daredevil #1
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Matthew Michael Murdock
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-hawkeye"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Hawkeye"
              className="card-img-top"
              src="/assets/marvel-hawkeye.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hawkeye
              </h5>
              <p
                className="card-text"
              >
                Tales of Suspense #57
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Clinton Francis Barton
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-cyclops"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Cyclops"
              className="card-img-top"
              src="/assets/marvel-cyclops.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Cyclops
              </h5>
              <p
                className="card-text"
              >
                X-Men #1
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Scott Summers
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-silver"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Silver Surfer"
              className="card-img-top"
              src="/assets/marvel-silver.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Silver Surfer
              </h5>
              <p
                className="card-text"
              >
                The Fantastic Four #48
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Norrin Radd
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en DashboardRoutes.js should show correctly /search 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current="page"
          className="nav-item nav-linkactive"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end container"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link"
        >
          Sara
        </span>
        <button
          className="nav-item nav-link btn btn-outline-warning"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Búsqueda
      </h1>
       
      <hr />
      <div
        className="row"
      >
        <div
          className="col-5"
        />
        <form
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            className="form-control"
            name="searchElement"
            onChange={[Function]}
            placeholder="buscar un héroe"
            type="text"
            value=""
          />
          <button
            className="btn btn-primary mt-5"
            onClick={[Function]}
            type="submit"
          >
            Buscar...
          </button>
        </form>
      </div>
      <div
        className="col-7 w-100 h-100"
      >
        <h4>
          Resultados
        </h4>
        <hr />
        <div
          className="row"
        >
          <div
            className="alert alert-info"
          >
            Buscar un héroe
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
